src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -285,6 +285,16 @@ public async Task InitializeAsync()
             _logger.LogTrace($"Done inferring Sql database schema in {timer.ElapsedMilliseconds}ms.");
         }
 
+        /// <inheritdoc />
+        public void InitializeAsync(
+            Dictionary<string, DatabaseObject> entityToDatabaseObject,
+            Dictionary<string, string> graphQLStoredProcedureExposedNameToEntityNameMap)
+        {
+            EntityToDatabaseObject = entityToDatabaseObject ?? EntityToDatabaseObject;
+            GraphQLStoredProcedureExposedNameToEntityNameMap = graphQLStoredProcedureExposedNameToEntityNameMap ?? GraphQLStoredProcedureExposedNameToEntityNameMap;
```

Comment:
Could you add a test case for this initializeAsync as well? Also I think another test that would be useful would be query execution scenario as this is eventually going to be used to execute queries. You could use existing integration tests for query execution and replace the old initializeasync with the call to this.

Reply:
Created a work item : "https://dev.azure.com/powerbi/AppDev/_backlogs/backlog/AppDev%20Team/Epics?workitem=1356906:"